### PR TITLE
🙌🏽 Map labels

### LINKED ISSUE #76: Map labels
## Description

Add labels to the locations on the map.

image

Relevant component: `components/map/Location.js`
Branch name: `map-labels`

## Learning objectives

- Context
- Centering

## Todo

- Style and rework Location.js to optionally take a label via a label prop
- This label should be centered below the blue circle
- Take a look at Slider.js ([data-reach-slider-marker] div) for how to center text under a small div
- Map.js should take a boolean prop, showLabels
- Add a SettingsContext in `ui/`. This should be modeled off of `SearchContext`
- Add a `showLabels` field in the object
- MapPage should pass the showLabels boolean in SettingsContext to Map
- Map should grab the appropriate field from the locations prop (for the common name) to pass as the label prop in Location

## Useful links

- Swagger for finding which field is the common name

### PR DESCRIPTION
**Pull request by vasuchal**
## Description
Displays Map Labels under each location based on the boolean showLabels prop. The labels show the common name for the trees. 

Fixes #76 

## Screenshots
Screen Shot 2021-04-14 at 8 12 37 PM

### CODE REVIEW

**Review by Archna-1 — APPROVED** [review]
LGTM! :)

**Review by code-arman — CHANGES_REQUESTED** [review]
Great work. The code looks good, but I feel like the experience/design itself needs more thought before the feature is merged in

It looks really good when the user is zoomed in really close, but other than that, it feels kinda weird. I wonder if Siraj would have ideas on a way to make it look less bad at mid level zoom.

-- Oops, just realize you mentioned this on slack as well, but i will still keep this comment i guesss

Screen Shot 2021-04-16 at 11 51 43 AM
Screen Shot 2021-04-16 at 11 47 59 AM
Screen Shot 2021-04-16 at 11 47 41 AM

File: src/components/map/Location.js [hunk 1]
```diff
@@ -21,9 +22,30 @@ const Location = styled.button`
...
+  /* Prevents line breaks */
+  white-space: nowrap;
+  z-index: 2;
+`
+const Location = ({ label }) => (
+  <>
+    <LocationContainer></LocationContainer>
+    <StyledDiv>{label}</StyledDiv>
+  </>
+)
```
Comment (Qrtn) [inline src/components/map/Location.js:45]: Does onClick still work? Does it get passed through the props?

File: src/components/map/Location.js [hunk 2]
```diff
@@ -21,9 +22,30 @@ const Location = styled.button`
     outline: none;
   }
 `
-
+const StyledDiv = styled.div`
```
Comment (code-arman) [inline src/components/map/Location.js:25]: Name this better plz

**Review by Qrtn — APPROVED** [review]
Nice work Vasu! Great job figuring out the tricky CSS.

**Pull request merged**